ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/VRF.hs

```diff
@@ -0,0 +1,126 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures             #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
+
+module Ouroboros.Consensus.Protocol.Praos.VRF (
```

Comment:
Lets include a reference here to the tech report on the VRF range-extension. It's now in the IOHK library: https://iohk.io/en/research/library/papers/on-uc-secure-range-extension-and-batch-verification-for-ecvrf/